Minor changes to support `FrameIteratorIndices` and `Timestep` in `_run_parallel` in `fingerprint.py`

wehs7661 opened the pull request:
Hi,
Thank you for your work on ProLIF. I've made a small change to support `FrameIteratorIndices` in `_run_parallel` to better handle non-contiguous MD trajectories.

Specifically, the current implementation of `_run_parallel` in `fingerprint.py` only supports trajectories as `XTCReader` objects (where the attribute `n_frames` is available) and `MDAnalysis.coordinates.base.FrameIteratorSliced ` objects (where the attributes `start`, `stop`, and `step` are available). This PR adds support for `MDAnalysis.coordinates.base.FrameIteratorIndices`, which can be useful for one to use ProLIF to analyze certain frames from an MD trajectory (e.g., frames in a cluster, which are not necessarily contiguous).

cbouysset commented:
Hi @wehs7661 and thanks a lot for contributing to ProLIF 🎉 

This is a really nice addition to better cover different trajectory indexing patterns, thanks for sharing it 👍 
Would you be okay with adding a test for non-contiguous frames?

I'll enable CI pipelines in the meantime

wehs7661 commented:
Hi @cbouysset, definitely, I have modified `test_run` in `test_fingerprint.py`. I also added changes to `fingerprint.py` to support `MDAnalysis.coordinates.timestep.Timestep` so that the user can run analysis for a single frame if desired. This is also tested in `test_run`. 

Please let me know if anything else is needed. Thanks!

cbouy commented:
I'll rebase your branch and fix the formatting issues don't worry about it. I'll also add the single frame change to running in serial.

cbouy commented:
Thank you for the contribution @wehs7661 !
